### DISCUSSION TITLE
Github Job summary working in windows runner

### DISCUSSION THREAD

**Discussion by karthik-philips-ta** [community/community#28294] [Actions]
I am trying to enable job summary ,job executes successfully but there is no job summary updated ,I suspect the issue may due to I am running the job in windows runner and there is different syntax to be used. Kindly help to get the right syntax or solution to the problem
=====================================================
      - name: Update Job Summary
        id: parsed-data-job
        run: |
              import pandas as pd
              url = "D:/coverity_action/_work/coverity-windows-action/coverity-windows-action/test/htmlcov/index.html"
              dfs = pd.read_html(url)
              L = str(dfs)
              print(str(dfs))
              print(f"""::set-output name=parsed-data::""" + str(dfs))
              file1 = open("coverity_coverage.txt","w")
              file1.writelines(L)
              file1.close()
        working-directory: ${{ github.workspace }}\test\     
        shell: python

      - name: job summary
        run: |
              echo "Coverity coverage below-" >> $GITHUB_STEP_SUMMARY
              echo ${{ steps.parsed-data-job.outputs.parsed-data }} >> $GITHUB_STEP_SUMMARY
===============================================================

also updated below didnt work

      - name: job summary
        run: |
              $GITHUB_STEP_SUMMARY = "Coverity coverage below"

**Comment by airtower-luna** [comment]
Considering you're already using Python, the easiest thing is to write to `GITHUB_STEP_SUMMARY` directly from there. All `echo "text" >> $GITHUB_STEP_SUMMARY` does (on platforms where it's supported) is append the given text (plus a newline) to the file named by the `GITHUB_STEP_SUMMARY` environment variable.

So inside your "Update Job Summary" step you could do something like this:
```py
import os
# ... create your "dfs", etc. ...
with open(os.environ['GITHUB_STEP_SUMMARY'], 'a') as fh:
    print('Coverity coverage below-', file=fh)
    print(str(dfs), file=fh)
```

Mode `'a'` for `open()` means "append" mode, so anything written to the file will be appended to existing content if any.
